cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py

```diff
@@ -50,16 +45,31 @@
 from charmhelpers.core.host import service_stop, service_restart
 from charmhelpers.contrib.charmsupport import nrpe
 
+from charms.layer.kubernetes_common import kubeclientconfig_path
+from charms.layer.kubernetes_common import migrate_resource_checksums
+from charms.layer.kubernetes_common import check_resources_for_upgrade_needed
+from charms.layer.kubernetes_common import calculate_and_store_resource_checksums # noqa
+from charms.layer.kubernetes_common import get_ingress_address
+from charms.layer.kubernetes_common import create_kubeconfig
+from charms.layer.kubernetes_common import kubectl_manifest, kubectl_success
+from charms.layer.kubernetes_common import kubectl
+from charms.layer.kubernetes_common import ApplyNodeLabelFailed
+from charms.layer.kubernetes_common import arch, get_node_name
+from charms.layer.kubernetes_common import gethostname
+from charms.layer.kubernetes_common import set_label, remove_label
+from charms.layer.kubernetes_common import configure_kubernetes_service
```

Comment:
I can't help but feel like many of these are being moved to `layer-kubernetes-common` prematurely.

`get_node_name` gets a node name for the current machine, meaning it can only be used by kubernetes-worker.

The label functions like `set_label`, `remove_label`, and `ApplyNodeLabelFailed` are only used by kubernetes-worker, and they call `get_node_name` anyway so it's not like anyone else could run it.

My specific concern is that moving this code makes it more distant and harder to follow - it's now part of a different file, in a different layer, which is pulled in at build time and may or may not be the version of the layer you would expect.

If it's not really common code, why move it?

Reply:
Looking into this more brings up the kube-proxy on master question. When we configure kube-proxy we give it the node name of the node. I think get_node_name has to stay, but the point is still very valid. Functions like set_label and remove_label are generic and could be used across charms, but currently are not and don't need to be in the shared lib just in case someone wants to use them. I'll audit the functions.

Reply:
You can run kube-proxy without giving it a node name. It will just default to using the hostname. It doesn't really matter because it's going to fail to find the node anyway (since no kubelet has registered with the name) but should continue to run with the basic functionality we need.

Though I guess you could argue that master and worker should configure kube-proxy the same way. That makes sense.

In general I would feel more comfortable if we didn't move stuff over to layer-kubernetes-common until it is actively being used in two places (i.e. wait to move `get_node_name` until both master and worker are running kube-proxy) but given that you've already moved it, and can make a case for its use in the near future, I'm OK with leaving it if you think that's best.